### ISSUE TITLE
Unreal 4.22 (preview 7) fails during ue4-full build

### ISSUE THREAD

**Issue by Clockwork-Muse** [adamrehn/ue4-docker#24]
Output of the `ue4-docker info` command:

```
ue4-docker version:         0.0.26
Operating system:           Linux (Ubuntu 18.04.2 LTS, 4.15.0-46-generic)
Docker daemon version:      18.09.4
NVIDIA Docker supported:    Yes
Maximum image size:         No limit detected
Total system memory:        31.36 GiB
Number of processors:       6 physical, 12 logical

```

Attempting to build the ue4-full image fails.  Unreal [has optimized/changed the build tool](https://forums.unrealengine.com/unreal-engine/announcements-and-releases/1583659-unreal-engine-4-22-preview), and I'm guessing their changes throw the current tooling out of whack.

Output of attempting to build the ue4-full image:
```
Step 9/25 : RUN ue4 conan generate && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''
 ---> Running in 515668b31656
Removing the "ue4" Conan profile if it already exists...
Removing any previous versions of profile base packages...

Detected clang version 7.0.1:
/home/ue4/UnrealEngine/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v13_clang-7.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang

Detected clang++ version 7.0.1:
/home/ue4/UnrealEngine/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v13_clang-7.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang++

Creating "ue4" Conan profile using autodetected settings...
Installing profile base packages...
Retrieving thirdparty library list from UBT...
Using user-specified engine root: /home/ue4/UnrealEngine
Traceback (most recent call last):
  File "/usr/local/bin/ue4", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/ue4cli/cli.py", line 216, in main
    SUPPORTED_COMMANDS[command]['action'](manager, args)
  File "/usr/local/lib/python3.6/dist-packages/conan_ue4cli/main.py", line 34, in main
    SUBCOMMANDS[subcommand]['function'](manager, args[1:])
  File "/usr/local/lib/python3.6/dist-packages/conan_ue4cli/generate.py", line 197, in generate
    libs = [lib for lib in manager.listThirdPartyLibs() if lib != 'libc++']
  File "/usr/local/lib/python3.6/dist-packages/ue4cli/UnrealManagerBase.py", line 178, in listThirdPartyLibs
    return interrogator.list(self.getPlatformIdentifier(), configuration, self._getLibraryOverrides())
  File "/usr/local/lib/python3.6/dist-packages/ue4cli/UE4BuildInterrogator.py", line 19, in list
    modules = self._getThirdPartyLibs(platformIdentifier, configuration)
  File "/usr/local/lib/python3.6/dist-packages/ue4cli/UE4BuildInterrogator.py", line 153, in _getThirdPartyLibs
    result = json.loads(Utility.readFile(jsonFile))
  File "/usr/local/lib/python3.6/dist-packages/ue4cli/Utility.py", line 30, in readFile
    with open(filename, 'rb') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp0egyj11y/ubt_output.json'
The command '/bin/sh -c ue4 conan generate && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''' returned a non-zero code: 1
[ue4-docker build] Error: failed to build image "adamrehn/ue4-full:custom-4.22.0-preview-7-cudagl10.0".
```

**Comment by adamrehn** [comment]
Thanks for reporting this! Yes, the JSON export mode used by ue4cli [has indeed been touched by the recent UBT refactor](https://github.com/EpicGames/UnrealEngine/blob/4.22.0-preview-7/Engine/Source/Programs/UnrealBuildTool/Modes/JsonExportMode.cs). What's rather concerning is that Epic appear to have forgotten to expose this mode via the updated command-line arguments, which means I can't update ue4cli until this regression is fixed. I've [submitted a pull request with the fix](https://github.com/EpicGames/UnrealEngine/pull/5658), which is actually just a single line of code. The moment it's merged, I'll update ue4cli accordingly, thus fixing the build of the `ue4-full` image.

**Comment by adamrehn** [comment]
Good news: it turns out that I misunderstood the exact usage of the new UBT command-line arguments, and JSON export mode is indeed exposed by UBT in 4.22 after all. This issue is now fixed in ue4cli version 0.0.29, which will use the appropriate arguments based on the detected Engine version. This will allow the `ue4-full` image to be built for 4.22.
